### ISSUE TITLE
 Ubuntu 18.04's ibus package breaks password fields in Firefox (by lowering & raising window whenever they're focused)

### ISSUE THREAD

**Issue by dholbert** [ibus/ibus#2002]
See also bugs filed on this issue elsewhere in Mozilla and Ubuntu's bug trackers:
  https://bugs.launchpad.net/ubuntu/+source/ibus/+bug/1765304
  https://bugzilla.mozilla.org/show_bug.cgi?id=1451466
(though I think this'll likely be the "main bug", since I'm pretty sure the proximal change/regression here is in ibus).

Which distribution and version?: 
Ubuntu 18.04 (prerelease, currently "beta")

Which desktop environment and version?: 
$XDG_CURRENT_DESKTOP="ubuntu:GNOME".  My Ubuntu packages for "gnome-shell" and (metapackage) "gnome" have version "3.28.0-0ubuntu5" and "1:3.22+9", respectively.

Which session type?: 
x11

Which application and version?: 
Firefox (current release, 59, as well as Nightly 61 from https://nightly.mozilla.org/ )

IBus version?: 
IBus 1.5.17

Issue description:
tl;dr: starting in Ubuntu 18.04, "ibus" seems to lower and raise Firefox, whenever the user tries to focus an <input type="password"> field on a web page. This is not a regression in Firefox, as the same Firefox version runs just fine in Ubuntu 17.10 -- it seems to be a bug in ibus, because it only happens (in 18.04) when the environmental variable GTK_IM_MODULE is at its default value (which is "ibus"). It had this same value in 17.10 but the bug didn't happen there; hence, I'm guessing there was a change in ibus (or a related package) that caused this breakage / lowering+raising.

This is pretty bad, because in some cases, the focus-loss can mean users are simply unable to fill in password fields that previously were working fine. (see "ACTUAL RESULTS" below)

STEPS TO REPRODUCE, method 1, via "BitWarden" Firefox extension:
 1. Start Firefox. (fresh profile if you like)
 2. Install BitWarden from https://addons.mozilla.org/en-US/firefox/addon/bitwarden-password-manager/
 3. Click the BitWarden toolbar icon to spawn a menu-pane.
 4. Click "Log in" at the bottom of that pane.
 5. Click the "Master Password" field.

ACTUAL RESULTS:
 The menu pane disappears as soon as the Master Password field receives focus.
EXPECTED RESULTS:
 I should be able to type in a password; pane shouldn't disappear.

STEPS TO REPRODUCE, method 2, via Reddit:
 1. Visit https://www.reddit.com/r/firefox/ (or any reddit page)
 2. Click "Log in or sign up in seconds" at the extreme upper right of the page.
      (click the "Log in" part of that sentence)
 3. Try to click the password field (or "tab" into it) and type in some text.

ACTUAL RESULTS:
 Try as you might, the password field never receives focus.
EXPECTED RESULTS:
 Password field should accept focus & let me type text into it.

FURTHER NOTES:
 - This is new in Ubuntu 17.10 (regardless of which Firefox version I'm using; I've tested the same Mozilla-provided Firefox tarball in both Ubuntu versions, and I only get this issue in 18.04)
 - This only happens when I have environmental variable GTK_IM_MODULE=ibus . The issue goes away if I unset that env var, or set it to the empty string.
 - Hence, I suspect this is a regression between the ibus versions between

Can you reproduce your problem when you restart ibus-daemon? (yes / no): 
(Run `ibus exit` and `ibus-daemon --xim &`)
Do you see any errors when you run ibus-daemon with the verbose option?:
(Run `ibus-daemon --xim --verbose &` and look at the output when you encounter your problem.)
The bug *does* still happen when I run those commands, though I'm not sure the first command is successfully making ibus exit (or maybe it autorestarts on its own), because when I run "ibus-daemon --xim", it says: "current session already has an ibus-daemon."

Can you reproduce your problem with a new user account instead of the current your account? (yes / no): Yes.

**Comment by fujiwarat** [comment]
![bitwarden-fujiwara](https://user-images.githubusercontent.com/105929/39030021-26d16d68-449a-11e8-8403-441defe5c5c8.png)

I cannot reproduce your problem with ibus 1.5.17 & firefox 59.0.1 in Fedora 28 and 27.
Seems your issue is Ubuntu specific.
I'm not sure if I can get time for Ubuntu. If you could test in Fedora LIVE DVD, it would be great.

```
% sudo dnf install ibus
% ibus-daemon --xim --verbose &
% env GTK_IM_MODULE=ibus firefox
```

> because when I run "ibus-daemon --xim", it says: "current session already has an ibus-daemon."

Right. gnome-shell restarts ibus-daemon until you rename it. Or maybe you could check the syslog.

**Comment by dholbert** [comment]
Indeed, I cannot reproduce in Fedora 28 (where `ibus version` says 1.5.18, only slightly newer than what I've got in Ubuntu). So this does seem Ubuntu-specific.

**Comment by dholbert** [comment]
Also, I did end up being able to run `ibus-daemon --xim --verbose` on Ubuntu (I just had to run it *immediately* after quitting, like so: `ibus exit && ibus-daemon --xim --verbose`)

Sadly, though, no logging appears at the moment when I trigger this bug.  The only logging I see is the following two startup warnings (which appear *immediately* when I start ibus-daemon):

> (ibus-ui-gtk3:3955): IBUS-WARNING **: 00:59:43.205: panel.vala:308: If you launch KDE5 on xterm, export XDG_CURRENT_DESKTOP=KDE before launch KDE5.
> 
> (ibus-ui-gtk3:3955): IBUS-WARNING **: 00:59:43.240: ibus_bus_call_sync: org.freedesktop.DBus.Properties.Get: GDBus.Error:org.freedesktop.DBus.Error.Failed: No global engine.

I didn't see those on Fedora... Any chance those are related somehow?)

**Comment by dholbert** [comment]
(Note: I don't know what the mention of KDE is about -- I'm definitely not running KDE.)

**Comment by fujiwarat** [comment]
Ah, I sorry. I just remembered:
https://bugzilla.redhat.com/show_bug.cgi?id=1552668

Please update mutter & gnome-shell

The above two warnings are expected messages and no problems.

**Comment by fujiwarat** [comment]
Ah, your issue is firefox. Probably it does not use mutter.

**Comment by dholbert** [comment]
And RE https://bugzilla.redhat.com/show_bug.cgi?id=1552668 : that does sound related, though I confess I don't understand how the pieces fit together clearly enough to see what the upshot is (and I'm not sure whether [your later comment](https://github.com/ibus/ibus/issues/2002#issuecomment-383032415) means there's something more to be done on the Firefox side, too).  Do you think the mutter fix discussed in that Red Hat bug tracker might fix this bug, too? (Looks like that fix is in mutter 3.28.1, so I imagine that'd be what I'd watch for.... it seems I have an Ubuntu mutter package with version "3.28.0-2" right now)

**Comment by fujiwarat** [comment]
I'd also ask you if your problem is the plugin of bitwarden only.
It would be good for you to try to access the password field in "https://mail.google.com" in Ubuntu.

> I'm using the latest available in Ubuntu right now

And then you may need to build mutter & gnome-shell by yourself.

> Would you mind elaborating a bit? 

I mean I could not find any links to mutter in /usr/lib*/firefox/* with `ldd` command and I wrote firefox does not use mutter. But I don't know how the plugins connect to the libraries so I think it still make senses to try 3.28.1 in Ubuntu.
Or probably I can downgrade mutter & gnome-shell to 3.28.0 in Fedora 28.
I know 3.26.0 is no problem in Fedora 27.

**Comment by dholbert** [comment]
> I'd also ask you if your problem is the plugin of bitwarden only.

It is not bitwarden only. As noted on the [Firefox version of this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1451466), this reproduces even on an extremly-basic testcase (like the one attached there) that has a `<input type="password">`, as can be observed via the logging noted there -- when the passord field is focused, something steals focus away from Firefox and then immediately returns it.

For a typical page, that's not catastrophic, but for BitWarden, it means the menu-popup disappears  before you get a chance to type in your password.  Similar things happen for a particular Reddit login page, too.

>  I think it still make senses to try 3.28.1 in Ubuntu.

I agree! Looks like if I tick the ubuntu 'bionic-proposed' checkbox in the repository sources dialog, it gives me a mutter package with version 3.28.1-1. I'm testing that and I'll report back...

**Comment by dholbert** [comment]
Darn, I can still reproduce the bug, after installing mutter 3.28.1-1 from bionic-proposed and rebooting. :-(  So mutter 3.28.1 does not seem to fix this.

**Comment by fujiwarat** [comment]
> Darn, I can still reproduce the bug, after installing mutter 3.28.1-1 from bionic-proposed and rebooting. :-( So mutter 3.28.1 does not seem to fix this.

Sorry for the result :(.
I guess firefox relative libraries are changed for the password logic.
BTW I use the bitwarden plugin version 1.26.3

>  This is not a regression in Firefox, as the same Firefox version runs just fine in Ubuntu 17.10

Which version of ibus does Ubuntu 17.10 use?

**Comment by dholbert** [comment]
(I can't personally try that for a few weeks -- sorry! I'm just about to take some vacation and will not have a computer with me, and I don't have enough time to get an ibus build environment configured before I go. If this is still an issue when I return, I'm happy to give that a shot though.)

**Comment by Raytar** [comment]
Hi, I just downloaded the following packages:

 * [gir1.2-ibus-1.0_1.5.14-2ubuntu1_amd64.deb](https://packages.ubuntu.com/artful/ibus)
 * [ibus_1.5.14-2ubuntu1_amd64.deb](https://packages.ubuntu.com/artful/gir1.2-ibus-1.0)
 * [ibus-gtk_1.5.14-2ubuntu1_amd64.deb](https://packages.ubuntu.com/artful/ibus-gtk)
 * [ibus-gtk3_1.5.14-2ubuntu1_amd64.deb](https://packages.ubuntu.com/artful/ibus-gtk3)

from [packages.ubuntu.com](https://packages.ubuntu.com) and installed them in an Ubuntu 18.04 virtual machine, and with these versions of the packages the issue is gone.

**Comment by fujiwarat** [comment]
> It seems that ibus-gtk3 is the only one that is needs to be downgraded for the issue to disappear.

OK, probably I think it's good to check if `g_signal_emit()` is called in `ibus/client/gtk3/ibusimcontext.c:ibus_im_context_commit_event()` in ibus 1.5.17.
https://github.com/ibus/ibus/blob/master/client/gtk2/ibusimcontext.c#L289

**Comment by Raytar** [comment]
I guess so, the file from the Ubuntu source is identical to master.

**Comment by Raytar** [comment]
I have tried building ibus 1.5.7 without the ubuntu patches, which did not solve the problem. I have also tried building ibus 1.5.8 from debian and that did not solve the problem either.

**Comment by fujiwarat** [comment]
I'd ask you if you can debug why `g_signal_emit()` is not called above when you type the password characters.

**Comment by Raytar** [comment]
Here is a backtrace when typing in a password field in a simple html page: [ubuntu_ibus.txt](https://github.com/ibus/ibus/files/1951196/ubuntu_ibus.txt)

**Comment by fujiwarat** [comment]
> Here is a backtrace when typing in a password field in a simple html page

Are you able to see the same backtrace when you type password characters in BitWarden?
My backtrace is on BitWarden in Fedora.

**Comment by Raytar** [comment]
> Are you able to see the same backtrace when you type password characters in BitWarden?
My backtrace is on BitWarden in Fedora. 

That's the problem. Once I try to focus the password field, the login dialog disappears.
Here's a gif: 
![peek 2018-04-27 08-57](https://user-images.githubusercontent.com/10762082/39348879-2ff19ad0-49f9-11e8-8172-6a381ec209b2.gif)

**Comment by fujiwarat** [comment]
Thank you and it's an interesting result.
Are you able to get the stacktrace in "Email address" field in BitWarden?

Since the changes between ibus 1.5.14 and 1.5.17 are only 4 commits, I'd ask you to revert one by one by manual to find which patch effects your problem.
```
% git log client/gtk2/ibusimcontext.c
```

1. d784e04c38eeb069f9a8da8b30743f4463fa34c3
client/gtk2: Fix a build failure with GDK_MODIFIER_INTENT_NO_TEXT_INPUT

2. bbfb3d738b9d61d1eb0658a9ce56e3cd8c111ac4
client/gtk2: Do not send key events to GtkIMContextSimple

3. 35ce62474fa97a5460d72c360943700a413a07ae
Support the portal in the gtk im modules

4. 6a3301db85e77e0652f7e00894cce493b6a942f6
client/gtk2: include the scaling factor

To revert 1, you can delete #if and the lines between #else and #endif in ibus/clients/gtk3/ibusimcontext.c and runs `make` and replace ibus/clients/gtk3/.libs/im-ibus.so with /usr/lib64/gtk-3.0/3.0.0/immodules/im-ibus.so.

Or you can apply [patch1.txt](https://github.com/ibus/ibus/files/1954772/patch1.txt) instead of the manual edit:
```
# (build ibus 1.5.17 by yourself in ibus directory. Maybe apt-get source ibus and debuild -us -uc ?)
# cd ibus
# patch -p1 -b -z .1 < patch1.txt
# cd client/gtk3
# make
# mv  /usr/lib64/gtk-3.0/3.0.0/immodules/im-ibus.so  /usr/lib64/gtk-3.0/3.0.0/immodules/im-ibus.so.orig
# cp .lib/im-ibus.so /usr/lib64/gtk-3.0/3.0.0/immodules/im-ibus.so
```

To revert 2, you can replace `ibus_im_context_commit_event` with `gtk_im_context_filter_keypress` in ibus/client/gtk3/ibusimcontext.c and build im-ibus.so and replace it.

Or you can apply [patch2.txt](https://github.com/ibus/ibus/files/1954779/patch2.txt):
``patch -p1 -b -z .2 < patch2.txt``

To revert 3, you can replace `ibus_bus_new_async_client` with `ibus_bus_new_async` and replace `ibus_bus_get_service_name (_bus)` with `IBUS_SERVICE_IBUS` and build im-ibus.so and replace it.

Or you can apply [patch3.txt](https://github.com/ibus/ibus/files/1954794/patch3.txt):
``patch -p1 -b -z .3 < patch3.txt``

To revert 4, you can delete the lines of _set_rect_scale_factor_with_window() and build im-ibus.so and replace it.
Or you can apply [patch4.txt](https://github.com/ibus/ibus/files/1954799/patch4.txt)

**Comment by fujiwarat** [comment]
> It says xkb:us::eng.

OK, it's good to reply my questions in https://github.com/ibus/ibus/issues/2002#issuecomment-384937029 since you said your problem is not reproduced in 1.5.14 but in 1.5.17.

**Comment by Raytar** [comment]
Yes, sorry. Let me add some more information:
* In ubuntu 17.10, the [ibus-xx-f19-password](https://github.com/ibus/ibus/files/1961144/ibus-xx-f19-password.txt) patch was included. 
* In ubuntu 18.04, the patch is removed. 
* If I try to build ibus 1.5.14 on ubuntu 17.10 without this patch, then the problem happens on 17.10 too. 
* If I build ibus 1.5.17 on ubuntu 18.04 with this patch, then the issue is gone.

**Comment by fujiwarat** [comment]
OK, I see your point.
But ibus-xx-f19-password does not intend to succeed the focus-in event.
Can you investigate which line is failed in ibus_im_context_focus_in() of client/gtk3/ibusimcontext.c without ibus-xx-f19-password?
Probably you can use gdb or adding printf() in ibusimcontext.c ?

**Comment by subdavis** [comment]
Just wanted to drop in and say that is causing the exact same behavior in my Firefox extension, subdavis/Tusk, which also auto-focuses a password-type input field via `element.focus()` when it opens.

Thanks in advance to folks participating here.

**Comment by vaer-k** [comment]
I'm seeing this issue on Ubuntu 18.04 in both the Bitwarden extension and [Metamask](https://metamask.io/) extension.

**Comment by paul-mesnilgrente** [comment]
Did anybody investigate this bug? I have the same problem on my Ubuntu 18.04 / Firefox / LastPass or Bitwarden.

**Comment by dholbert** [comment]
>  and it's caused with Ubuntu desktop environment.

Actually, I've just seen reports that this *does* happen in Fedora as well! But apparently you have to be using X11 (I think wayland is the fedora default now, which is unaffected). See https://bugzilla.mozilla.org/show_bug.cgi?id=1405634#c0 for more

**Comment by dholbert** [comment]
@fujiwarat perhaps you can reproduce and debug locally if you switch your Fedora install to be using X11 instead of Wayland?

**Comment by fujiwarat** [comment]
@dholbert No, I cannot reproduce your problem in both Wayland and Xorg GNOME, XFCE in Fedora.

**Comment by beyera** [comment]
I came across a reference to this issue from Bugzilla. 

We have [1Password X](https://support.1password.com/getting-started-1password-x/) users running into this same problem. Before finding related issues [I created a simple Firefox extension](https://github.com/beyera/ff-password-focus-bug) which reproduces this issue to aid in troubleshooting. 

Thanks to everyone helping to get this resolved.

**Comment by revaldozen** [comment]
I have this problem too with addon extension windscribe and alldebrid.

**Comment by galvani** [comment]
I confirm this bug on IBus 1.5.17 and not only firefox but thunderbird password field in accounts setup. I have no extensions in browser nor in thunderbird. But are from ubuntu repositories.

**Comment by lauhayden** [comment]
Note that to reproduce on the Reddit login page, you have to go to the [old reddit design]( https://old.reddit.com/). Click 'Log in or Sign Up in seconds' in the top right.

A workaround for the LastPass FF Addon is to use the Ctrl+Alt+H shortcut key to attempt to access your Vault, that'll open the full-page login screen if you're not logged in. That password field works.

This bug is supremely annoying - I'm also experiencing this bug in other pages as well, such as [Sonarr](https://github.com/Sonarr/Sonarr) and [Radarr](https://github.com/Radarr/Radarr)'s password fields.

**Comment by rawkode** [comment]
Can confirm this is a problem on Arch with 1.5.18-1

**Comment by galvani** [comment]
I have just disabled the ibus and all is working fine. It's one of many problems I have had with ibus along the years.

**Comment by mgray88** [comment]
With Ubuntu 18.04/Firefox 60.0.1, switch to Wayland and the problem goes away

**Comment by fujiwarat** [comment]
@subdavis You can diisable ibus with `env GTK_IM_MODULE=gtk-im-context-simple firefox`.

**Comment by galvani** [comment]
@mgray88 yes, but you will loose you nvidia in case you have one.

**Comment by ardnaxele** [comment]
Hi Guy,
I went into the language menu of ubuntu and i saw a keyboard option that was"ibus". I change it by Xim and now everything works. Do you know what it change? My nvidia graphic card will not be impacted?
My ubuntu is in french the field i changed is "Système de saisie au clavier"

**Comment by drazisil** [comment]
@ardnaxele This snot not appear to correct the issue for me. Did you need a restart of anything?

**Comment by ardnaxele** [comment]
I restarted the computer. It don't work if i dont do it. I also clic on apply to all system before reboot.

**Comment by lucianoratamero** [comment]
for now, what I'm doing is to add a new `firefox-patched.desktop` file on `~/.local/share/applications/` with the contents of this gist: https://gist.github.com/lucianoratamero/81f0ef920a88a607e53b858af028d00e

**Comment by svtcalvin** [comment]
On Ubuntu 18.04 the problem is solved if you installed the available package by clicking the link below
[https://bugzilla.mozilla.org/attachment.cgi?id=8974077](https://bugzilla.mozilla.org/attachment.cgi?id=8974077)
Here is the origin of the .deb
https://bugzilla.mozilla.org/show_bug.cgi?id=1405634

**Comment by mjedmonds** [comment]
As mentioned by @ardnaxele, switching to XIM resolves the issue for me (on Xorg). For a more clear description: go to Language Support -> Keyboard input method system and select XIM instead of IBus. Restart/logout.

I can't attest to any negative side effects from this, but I have experience none thus far. XIM is installed with 18.04 as well, so I don't *think* this should cause any problems. I'm using 1Password X, but hopefully this fixes the issue for all password/input managers.

**Comment by favadi** [comment]
> I can't attest to any negative side effects from this,

The negative side is I can't type in my native language any more :).

I've installed  Fedora 28 and the problem is reproducible by switching from Wayland to X11..

**Comment by fujiwarat** [comment]
> I've installed Fedora 28 and the problem is reproducible by switching from Wayland to X11..

Thank you. Now I can reproduce your problem in GNOME Xorg but not reproduce in XFCE.
I will investigate this issue.

**Comment by fujiwarat** [comment]
The problem happens in GNOME Xorg only but not other desktops; XFCE4, LXDE and so on.

**Comment by goodevilgenius** [comment]
I have this problem in Fedora 28, using GNOME 3.28.3 on Xorg. I have the issue with two password manager extensions I use.

**Comment by goodevilgenius** [comment]
Doesn't happen with Wayland.

**Comment by fujiwarat** [comment]
I added a workaround in ibus clients with the patch.

https://github.com/fujiwarat/ibus/commit/ca8ce8ba969ef679cf18ddf2872d31426b21a2db

    env IBUS_DISCARD_PASSWORD=1 firefox
    or
    export IBUS_DISCARD_PASSWORD_APPS='firefox,.*chrome.*'
    
    can discard typing characters on the password entries.

**Comment by MatzFan** [comment]
Workaround provided by @ardnaxele works for me in Ubuntu 18.04: Go to `Settings/Region & Language/Manage Input Languages/Keyboard input method system` change IBus to XIM & restart.

**Comment by dholbert** [comment]
@fujiwarat:
> IBUS_DISCARD_PASSWORD and IBUS_DISCARD_PASSWORD_APPS environment variables are available in ibus 1.5.19 or later.

Thanks! Do you have a suggested way for these env variables to be used to work around this?  (Would it be worth it for e.g. Ubuntu to set this variable globally (or globally-enough to be seen by the ibus service) in some package update, until [your gnome-shell patch](https://gitlab.gnome.org/GNOME/gnome-shell/issues/391) has been merged and shipped?  (Do you happen to know if this is something that they could set in the ibus package itself, or where the best place would be for them to set that variable if they were going to do it in an automated way via a package update?)

On the Mozilla/Firefox end: we're still seeing significant user pain from this issue in Firefox (password fields on [some](https://github.com/webcompat/web-bugs/issues/18134) [sites](https://github.com/webcompat/web-bugs/issues/18188) being literally unfocusable as a result of this bug, making Firefox unusable to log in to those sites).  I'm quite eager to know what the best action is that we can take in the near term that doesn't require individual per-user action to apply a local workaround.

**Comment by fujiwarat** [comment]
> Ubuntu to set this variable globally (or globally-enough to be seen by the ibus service) in some package update

Probably I think Ubuntu can modify [ibusimcontext.c:94](https://github.com/ibus/ibus/blob/master/client/gtk2/ibusimcontext.c#L94) with an internal patch

```
- static const gchar *_discard_password_apps = "";
+ static const gchar *_discard_password_apps = "firefox,.*chrome.*";
```

Or [ibus-xx-f19-password.txt](https://github.com/ibus/ibus/files/1961144/ibus-xx-f19-password.txt)

I'm not a fan of applying the workaround by default for any distributions because this issue needs to be reproduced by the maintainers of gnome-shell.
I also think the workaround would be applied only when the desktop is GNOME Xorg and application is the web browser because the workaround has a small side effect in gnome-shell's password lock.

**Comment by Multi-Dimensional** [comment]
The post from svtcalvin solved the issue for me, thanks!

> On Ubuntu 18.04 the problem is solved if you installed the available package by clicking the link below
https://bugzilla.mozilla.org/attachment.cgi?id=8974077
Here is the origin of the .deb
https://bugzilla.mozilla.org/show_bug.cgi?id=1405634

**Comment by JackPShafer** [comment]
> On Ubuntu 18.04 the problem is solved if you installed the available package by clicking the link below
> https://bugzilla.mozilla.org/attachment.cgi?id=8974077
> Here is the origin of the .deb
> https://bugzilla.mozilla.org/show_bug.cgi?id=1405634

This worked for me as well!

**Comment by aveeshkumar** [comment]
Thank you Thank you Thank you. I was about to install the snap version of firefox (which I really didnt want to)

Thank you all you gentlement for getting to the bottom of this, finding a solution and educating us real world mortals. A thousand cups of ubuntu for you !!

**Comment by fossterer** [comment]
> This is the proposed patch; https://gitlab.gnome.org/GNOME/gnome-shell/merge_requests/140/diffs?commit_id=94cca97ad599f1da9bc52d8ff5cb28df681a3538
> 
> I submitted the issue for gnome-shell; https://gitlab.gnome.org/GNOME/gnome-shell/issues/391
> 
> I see why [ibus-xx-f19-password](https://github.com/ibus/ibus/files/1961144/ibus-xx-f19-password.txt) can be a workaround from @Raytar .

@fujiwarat I +1'd your PR on gitlab. Can you tell who else to be moved to get this accepted?
I faced this issue in multiple websites (can't distinctly remember) so far and today in our company's one internal web application.

**Comment by fujiwarat** [comment]
@fossterer I think there is no time for GNOME 3.30 and the target is 3.32 and now is still the code freeze.

In ibus 1.5.19, IBUS_DISCARD_PASSWORD or IBUS_DISCARD_PASSWORD_APPS would be a workaround,
`env IBUS_DISCARD_PASSWORD=1 firefox` or `export IBUS_DISCARD_PASSWORD_APPS='firefox,.*chrome.*'`

**Comment by fossterer** [comment]
@fujiwarat Thanks for the response.
I'm running IBus 1.5.17 . Neither of the 2 commands above solved it for me.
I'm under the impression that this issue does not exist in 3.32.
Per your experience with the Gnome project, how long would you say it would take for Ubuntu to include this 3.32? Next release of Ubuntu?

**Comment by fujiwarat** [comment]
@fossterer Sorry, I cannot commit the schedule since I'm not the maintainer of gnome-shell.
I'm also not sure about Ubuntu schedule. After the fix is integrated in GNOME, probably you can ask Ubuntu to backport the fix.

Probably you may like to comment on gnome-shell/issues/391 to push the fix asap.

**Comment by ilovekuchen** [comment]
> On Ubuntu 18.04 the problem is solved if you installed the available package by clicking the link below
> https://bugzilla.mozilla.org/attachment.cgi?id=8974077
> Here is the origin of the .deb
> https://bugzilla.mozilla.org/show_bug.cgi?id=1405634

This worked for me too. should be in the starting post or somehwere on the wiki.

**Comment by Imfromthenet** [comment]
> > On Ubuntu 18.04 the problem is solved if you installed the available package by clicking the link below
> > https://bugzilla.mozilla.org/attachment.cgi?id=8974077
> > Here is the origin of the .deb
> > https://bugzilla.mozilla.org/show_bug.cgi?id=1405634
> 
> This worked for me too. should be in the starting post or somehwere on the wiki.

hi, I'm new here.
I have just freshly installed Ubuntu 18.04 and am facing the same issue. Installing the package you mentioned in the last comment did not fix the problem.

**Comment by Imfromthenet** [comment]
Temporary workaround: right click on extension icon, left click on Lastpass and enter your credentials into a newly opened tab.

https://forums.lastpass.com/viewtopic.php?f=12&t=300515

**Comment by fujiwarat** [comment]
@Imfromthenet You're right. The workaround does not fix your problem but My fix in gnome-shell resolves your issue.

**Comment by fujiwarat** [comment]
https://gitlab.gnome.org/GNOME/gnome-shell/commit/551e827841626cd8084daa2210b3bf60e5be96be

**Comment by berkes** [comment]
The patch has landed in gnome-shell, and is fixed there. Ubuntu has not yet backported as of writing.

The bug for ubuntu is here: https://bugs.launchpad.net/firefox/+bug/1794655

**Comment by rubo77** [comment]
> Go to `Settings/Region & Language/Manage Input Languages/Keyboard input method system` change IBus to XIM

How do you change the Settings in Firefox 76? There is no such settings any more

**Comment by fujiwarat** [comment]
> How do you change the Settings in Firefox 76? There is no such settings any more

It means `gnome-control-center region` command and an IM configuration tool in your distro.
But since the fix was delivered two years ago, it's good to update gnome-shell simply.
Or your issue is not this.
